python-main.js

```diff
@@ -1266,63 +1266,116 @@ function web_editor(config) {
-        console.log("An error occured whilst attempting to use WebUSB. Details of the error can be found below, and may be useful when trying to replicate and debug the error.");
+        console.log("An error occured whilst attempting to use WebUSB.");
+        console.log("Details of the error can be found below, and may be useful when trying to replicate and debug the error.");
         console.log(err);
+        console.trace()
 
-        // If there was an error clear dapwrapper
+        // If there was an error and quick flash is in use, then clear dapwrapper
         if(usePartialFlashing) {
-            window.dapwrapper = null;
-            window.previousDapWrapper = null;
+            if(window.dapwrapper) {
+                window.dapwrapper = null;
+            }
+
+            if(window.previousDapWrapper) {
+                window.previousDapWrapper = null;
+            }
         }
 
-        // Disconnect
+        // Disconnect from the microbit
         doDisconnect();
+       
+        var errorType;
+        var errorTitle;
+        var errorDescription;
 
-        // Error handler
-        $("#flashing-overlay-container").css("display", "flex");
-        $("#flashing-info").addClass('hidden');
+        // Determine type of error
+        switch(typeof err) {
+            case "object":
+                console.log("Caught in Promise or Error object");
+                // We might get Error objects as Promise rejection arguments
+                if (!err.message && err.promise && err.reason) {
+                    err = err.reason;
+                }
+                
+                // Determine error type
+                if (err.message === "No valid interfaces found.") {
+                    errorType = "update-req";
+                    errorTitle = err.message;
+                    errorDescription = config["translate"]["webusb"]["err"][errorType];
+                } else if (err.message === "Unable to claim interface.") {
+                    errorType = "clear-connect";
+                    errorTitle = err.message;
+                    errorDescription = config["translate"]["webusb"]["err"][errorType];
+                } else if (err.name === "device-disconnected") {
+                    errorType = "device-disconnected";
+                    errorTitle = err.message;
+                    // No additional message provided here, err.message is enough
+                    errorDescription = "";
+                } else {
+                    // Unhandled error. User will need to reconnect their micro:bit
+                    errorType = "reconnect-microbit";
+                    errorTitle = "WebUSB Error";
+                    errorDescription = config["translate"]["webusb"]["err"][errorType];
+                    if (usePartialFlashing) {
+                        errorDescription += '<br>' + config["translate"]["webusb"]["err"]["partial-flashing-disable"];
+                    } 
+                }
```

Comment:
As mentioned in a previous comment, we are swallowing error types, I think we should probably show the user.
For instance, if the code is too large, a microbit-fs error ends up here that is never shown to the user.
`Error: Oops! Could not load the code into the hex file, Not enough space for the main.py file.`

Reply:
How about something along these lines?

![Screenshot 2019-10-22 at 13 29 27](https://user-images.githubusercontent.com/32453267/67281569-0376c100-f4d0-11e9-8e8a-f22ed998aa26.png)

Reply:
Maybe with a much lighter background:
![image](https://user-images.githubusercontent.com/29712657/67282609-15edfc00-f4ca-11e9-844a-32a83d860f50.png)

```css
{
    margin: 16px 0px;
    padding: 12px;
    border-radius: 8px;
    background-color: #fafafa;
}
```

Reply:
👍